mne/cuda.py

```diff
@@ -19,6 +20,53 @@
 _cuda_capable = False
 
 
+def get_shared_mem(
```

Comment:
ok. I hear you on making this more DRY. I'm not sure I'm all the way there yet. 

`_share_cuda_mem` implies only calls from within `mne.cuda`.  This is fine. 

However, I think, when `_cuda_upload_rfft` is called by `fft_resample` `n_jobs' is not available.  `n_jobs` has been returned as 1 (i.e. 1 cpu job, to run parallel with cuda via gpu).   

And then the cleanest driest, would be for `mne.filter` to `_share_cuda_mem` (making it no longer a private call...) 

Or I'm missing the best place for `_share_cuda_mem` to be called.  I'll keep thinking on it. Still spinning up on the logic flow, I feel slow, lol. 

Further, `_share_cuda_mem(x, n_jobs)` ...could this just be `_share_cuda_mem(x)`.  bc, this is private, always called from within cuda, so assumed to be cuda?  or do what I did, and pass in "cuda" when we're already to a point of knowing?

Reply:
> However, I think, when _cuda_upload_rfft is called by fft_resample n_jobs' is not available.  n_jobs` has been returned as 1 (i.e. 1 cpu job, to run parallel with cuda via gpu).

If it's inside a call that we know is in the cuda-only path then you would just call `_share_cuda_mem(x, 'cuda')`

> And then the cleanest driest, would be for mne.filter to _share_cuda_mem (making it no longer a private call...)

This would still be private. Private vs public doesn't refer to existence in `mne.filter` vs `mne.cuda`, it refers to the leading underscore. `mne.filter._share_cuda_mem` and `mne.cuda._share_cuda_mem` are both private in the sense that they can be used inside our codebase however we need, but we can move them without any API deprecation period. (Users should never use a private attribute / method / function, i.e., one with a leading `_` in the name or in a so-named namespace.)

Reply:
copy.  I'm a recovering java -> python dev that gets confused about straddling the paradigms. 

that's a great explanation.  when you get a sec, take a look at how it looks now. 

The only better improvement I see is to call `_share_cuda_mem` before passing `W` and `x` into `mne.cuda` from `mne.filter`.  Thoughts?